internal/transport/handler_server.go

```diff
@@ -347,7 +355,7 @@ func (ht *serverHandlerTransport) WriteHeader(s *Stream, md metadata.MD) error {
 	return err
 }
 
-func (ht *serverHandlerTransport) HandleStreams(startStream func(*Stream)) {
+func (ht *serverHandlerTransport) HandleStreams(_ context.Context, startStream func(*Stream)) {
```

Comment:
Doesn't ignoring the context mess everything up?

Can grpc not pass the `http.Request.Context()` (with things like Peer added) so that it doesn't need to be ignored here?

Reply:
It was being ignored in master already, and it uses the requests Context (ctx := ht.req.Context()). I think this is the same functionality, and I don't think I want to add something to gRPC layer using the http.Request since it's a server handler transport specific concept.

Reply:
Discussed offline; decided to move peer creation to construction time and read it in gRPC before calling this.